assets/js/sign/validation.js

```diff
@@ -0,0 +1,71 @@
+import { errorDisplay, clearDisplay } from './ui.js';
+
+export function validateEmail(e, sign) {
+  const errorMsg = sign.errorMessages[0];
+  const email = sign.inputEmail.value;
+  const regex =
+    /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/;
+
+  if (email.length === 0) {
+    errorDisplay(e, errorMsg, '이메일을 입력해주세요');
+    return false;
+  } else if (!regex.test(email)) {
+    errorDisplay(e, errorMsg, '올바른 이메일 주소가 아닙니다.');
+    return false;
+  } else if (email == 'test@codeit.com' && sign.currentPage.includes('signup')) {
+    errorDisplay(e, errorMsg, '이미 사용 중인 이메일입니다.');
+    return false;
+  } else {
+    clearDisplay(e, errorMsg);
+    return true;
+  }
```

Comment:
이렇게 리팩터링해보시면 어떨까요?
조건문이 서로 독립적으로 실행되도록 변경함으로써, 각각의 조건에 대한 로직이 좀 더 명확해보이지 않으신가요? 😄 
```suggestion
if (email.length === 0) {
    errorDisplay(e, errorMsg, '이메일을 입력해주세요');
    return false;
}

if (!regex.test(email)) {
    errorDisplay(e, errorMsg, '올바른 이메일 주소가 아닙니다.');
    return false;
}

if (email === 'test@codeit.com' && sign.currentPage.includes('signup')) {
    errorDisplay(e, errorMsg, '이미 사용 중인 이메일입니다.');
    return false;
}

clearDisplay(e, errorMsg);
return true;
```

Reply:
아래와 같이 리팩토링하였습니다.

```javscript
if (email.length === 0) {
    return showErrorMessage(e, handleError('EMPTY_EMAIL_FIELD'));
  }

  if (!emailRegex.test(email)) {
    return showErrorMessage(e, handleError('INVALID_EMAIL'));
  }
  ```
  form에서 submit 할때 체크하는 true, false 부분은 `ui.js`에 추가해 두었습니다.
  
  ```javascript
  export function showErrorMessage(source, errorMessage) {
  const errorMessageElement = errorMessageElementFind(source);

  errorMessageElement.innerHTML = errorMessage;
  errorMessageElement.classList.add('input-error');

  return false;
}

export function clearErrorMessage(source) {
  const errorMessageElement = errorMessageElementFind(source);

  errorMessageElement.innerHTML = '';
  errorMessageElement.classList.remove('input-error');

  return true;
}
```